pelicun/base.py

```diff
@@ -626,29 +602,25 @@ def update_vals(
                 f'{primary_path}["{key}"]',
             )
         # if `primary[key]` is NOT a dictionary:
-        else:
-            # if `key` does not exist in `update`, we add it, with
-            # its corresponding value.
-            if key not in update_value:
-                update_value[key] = primary[key]
-            else:
-                # key exists in update and should be left alone,
-                # but we must check that it's not a dict here:
-                if isinstance(update_value[key], dict):
-                    raise ValueError(
-                        f'{update_path}["{key}"] '
-                        'should not map to a dictionary. '
-                        f'The specified value is '
-                        f'{update_path}["{key}"] = {update_value[key]}, but '
-                        f'the default value is '
-                        f'{primary_path}["{key}"] = {primary[key]}. '
-                        f'Please revise {update_path}["{key}"].'
-                    )
-    # pylint: enable=else-if-used
+        elif key not in update_value:
```

Comment:
Again, here I thought these comments were redundant. We should be adding comments to explain the logic, but not comments like
```
# create a list
my_list = []
# if the list is empty
if not my_list:
...
```
Such comments don't help. It's just clutter.

Reply:
Maybe. I see what you're saying with your example, but I feel these comments are potentially helpful for those who are less well versed in Python. In this particular case, it might be much easier for someone to read two lines of comments than to read the actual statement and make sure they understand it well. I tend to prefer to err on the side of having more comments than fewer because I don't think it hurts anyone.
For now, let's not bring these back, I can see your point, but please try to add comments whenever they make at least a little sense.

Reply:
Thanks for the feedback, I see your point, and I will try to do as you said when introducing new code.